bundler/spec/commands/init_spec.rb

```diff
@@ -7,6 +7,32 @@
     expect(bundled_app_gemfile).to be_file
   end
 
+  context "with a template with permission flags not matching current process umask" do
+    let(:template_file) do
+      gemfile = Bundler.preferred_gemfile_name
+      templates_dir.join(gemfile)
+    end
+
+    let(:target_dir) { bundled_app("init_permissions_test") }
+
+    before do
+      @old_chmod = File.stat(template_file).mode
+      FileUtils.chmod(@old_chmod | 0o111, template_file) # chmod +x
+    end
+
+    after do
+      FileUtils.chmod(@old_chmod, template_file)
+      FileUtils.rm_rf(target_dir)
```

Comment:
All these directories are removed automatically by our global teardown, so we generally don't do this particular specific cleanup.

Also, I would instead use an `around` hook, so that you don't need to keep the previous mode around in an instance variable.

Reply:
Hey @deivid-rodriguez! Sorry the delay for address this! Busy week.
I'll remove the cleanup, and replace `before`/`after` hooks with an `around`; completely forgot about it.

I'll force-push and mark this as resolved right after. Thank you for reviewing, and for the suggestions!